### ISSUE TITLE
Create Simple Telemetry/Command Interface For HAB

### ISSUE THREAD

**Issue by kb1lqd** [FaradayRF/Faraday-Software#145]
## Summary
The Rose-Hulman amateur radio club is preparing a high-altitude balloon with Faraday on-board and they would like a simple telemetry and command interface. The prior GUI used in https://faradayrf.com/simple-faraday-balloon-launch/ is not currently working with our new modular software. Due to time priority a simpler interface is acceptable.

## Problem Explanation
A simple user interface to:
* Receive and display telemetry live
* Basic telemetry ADC scaling functionality to scale analog sensors
  * Better left for Telemetry?
* Command Faraday for balloon mission

Since proxy and telemetry are Flask based a basic command prompt interface or simple webpage output is acceptable for telemetry viewing.

For commanding a simple command prompt interface is acceptable.

#### Software 
Faraday Master branch

#### Hardware

Standard Faraday hardware

# Supporting Information

**Comment by kb1lqc** [comment]
Looking at the code we flew with Faraday back in August 2016... it was intimately integrated with Proxy. No bueno. I really don't want to temporarily port that in. We've moved on quite readily.

It will likely be easy to provide an automatically updating jQuery based webpage to see telemetry. No frills. As for commanding, leaving that for command prompt reduces the jQuery and other work to integrate buttons with RESTful commands and therefor would simplify this more.

**Comment by kb1lqc** [comment]
I've pushed a few updates to my simpleui branch: https://github.com/kb1lqc/Faraday-Software/tree/simple_web_ui

I now have basic telemetry (subset) code updating to a webpage automatically using JQuery and javascript. This is a good step towards no frills telemetry. This requires the Telemetry CORS update on #148.

Below is a quick screencapture of the interface.

![simpleui](https://cloud.githubusercontent.com/assets/331540/24791070/9034ebdc-1b2d-11e7-9c20-08547871c025.gif)

**Comment by kb1lqc** [comment]
Current plan:
- [x] Scale telemetry in javascript to volts at the ADC 
- [x] Allow selection of station to be displayed
- [x] Add commanding support for basic IO commands
- [x] Provide buttons for commands
- [x] Provide simple Bootstrap styling

**Comment by kb1lqc** [comment]
Getting there. Boostrap styled UI with some javascript logic to display time and speed nicely as well as detect incorrect input data. Next up simple commanding!

![simpleui-bootstrap](https://cloud.githubusercontent.com/assets/331540/24848865/cb3ae87e-1d7d-11e7-83a2-a005679b2b9c.gif)

**Comment by kb1lqc** [comment]
Here's the latest UI, a bit rough but coming along. Needs a few of the Firmware and Telemetry issue tickets recently created to really polish up.

![simpleui-bootstrap2](https://cloud.githubusercontent.com/assets/331540/25034497/bf38e654-209b-11e7-88b8-d2afe733a3f0.gif)

**Comment by kb1lqc** [comment]
I've completed my initial run at this user interface. However, for it to fully be useable several issues must be addressed:
- Software #151 
- Software #153
- [Firmware #79](https://github.com/FaradayRF/Faraday-Firmware/issues/72) @kb1lqd 

In addition to these I'll also need to uncomment and reorganize some HTML code to bring these features online with the UI.

**Comment by kb1lqc** [comment]
I updated to properly grab IOstate and merged in a bunch of the latest master changes which changed a lot of uart folders and caused some merging problems with other merges. I've updated the UI to display all known data of interest.

@kb1lqd I cannot confirm the correct orientation of the bits for most IO as the mapping seems to be incorrect in firmware anyways per [Firmware #79](https://github.com/FaradayRF/Faraday-Firmware/issues/72). Any chance we can get this higher priority?

![simpleui-fullypopulatedled](https://cloud.githubusercontent.com/assets/331540/25082874/4b2a10b6-2308-11e7-9189-82858937f091.gif)

**Comment by kb1lqc** [comment]
Nearly done. Just finished up some styling items and added support for `grunt` checking of javascript and HTML (locally) with node.js using `jslint` and `htmllint`. @reillyeon getting there!

https://github.com/kb1lqc/Faraday-Software/tree/simple_web_ui/Applications/SimpleUI

**Comment by kb1lqc** [comment]
OK I updated `jslint` and `htmllint` as best as I could to conform to JQuery formatting per our `CONTRIBUTIONS.MD` file. I'm calling this one done. Eventually we'll revisit this when we automate any of this checking @reillyeon.

I also updated a GPIO bit order error and that fixed a bunch of the GPIO, however, I still can't control or see telemetry that makes sense from GPIO0, GPIO1, or GPIO2:
![image](https://cloud.githubusercontent.com/assets/331540/25168328/f0b32a82-2497-11e7-83c5-31f6355a9744.png)

I'll probably mark this as a todo update/issue to look into for firmware or software wherever it's located and pull this in on a PR @kb1lqd . That way we get most of the UI into use as we investigate that issue since it adds a lot of value to the project as it is now.
